### PR TITLE
fix: cache http wait request body

### LINKED ISSUE #338: HTTP Wait Strategy: POST body missing on subsequent calls
**Describe the bug**

HTTP POST body is only sent on the first call, and all subsequent calls are made with an empty body.

**To Reproduce**
Setup a `NewHTTPStrategy` using `http.MethodPost`, and include a `WithBody`.
```go
wait.NewHTTPStrategy("/api/v1/endpoint").
	WithPort("80/tcp").
	WithResponseMatcher(func(body io.Reader) bool {
		data, err := ioutil.ReadAll(body)
		return err == nil && bytes.Contains(data, []byte("OK"))
	}).
	WithPollInterval(time.Second).
	WithMethod(http.MethodPost).
	WithBody(bytes.NewReader([]byte(`{"foo": "bar"})))
```

In order for the bug to occur, the first POST needs to fail with an invalid HTTP code.

Example output from the web service receiving the POST calls:
```
2021/08/04 02:08:33 got body: '{"foo": "bar"}'
2021/08/04 02:08:34 got body: ''
2021/08/04 02:08:35 got body: ''
2021/08/04 02:08:36 got body: ''
2021/08/04 02:08:37 got body: ''
2021/08/04 02:08:38 got body: ''
```

Notice how the first call contains the body, and every one second after another call is received with an empty body.

**Expected behavior**
Every poll in the `WaitInterval` contains a POST with the same exact body.

**info**
`testcontainers-go  version  0.11.1`

**Additional context**
Here is the offending code (i believe). The `ws.Body` is emptied on first call. It needs to be cloned or cached, etc.

https://github.com/testcontainers/testcontainers-go/blob/b5bf62f1dd2273aad5cb0842ed606d043d3465f2/wait/http.go#L187-L196

### LINKED ISSUE #341: FAIL: Test_MultipleContainersInTheNewNetwork
**Describe the bug**
Failing unit test.

```
=== RUN   Test_MultipleContainersInTheNewNetwork
2021/08/04 11:03:06 Starting container id: 215c2d03fce2 image: quay.io/testcontainers/ryuk:0.2.3
2021/08/04 11:03:06 Waiting for container id 215c2d03fce2 image: quay.io/testcontainers/ryuk:0.2.3
2021/08/04 11:03:06 Container is ready id: 215c2d03fce2 image: quay.io/testcontainers/ryuk:0.2.3
2021/08/04 11:03:06 Starting container id: b4bfa770dce2 image: postgres:12
2021/08/04 11:03:07 Waiting for container id b4bfa770dce2 image: postgres:12
2021/08/04 11:03:08 Container is ready id: b4bfa770dce2 image: postgres:12
2021/08/04 11:03:08 Starting container id: 85c4278deffc image: rabbitmq:management-alpine
2021/08/04 11:03:08 Waiting for container id 85c4278deffc image: rabbitmq:management-alpine
    network_test.go:99: failed to start container: /bin/sh command not executable
--- FAIL: Test_MultipleContainersInTheNewNetwork (3.49s)

FAIL

Process finished with the exit code 1
```

**Expected behavior**
Test should pass.

**Additional context**
Error is due to a deprecation of the latest rabbitmq:

```
testcontainers-go$ docker logs d5d0643585eb
error: RABBITMQ_DEFAULT_PASS is set but deprecated
error: RABBITMQ_DEFAULT_USER is set but deprecated
error: RABBITMQ_ERLANG_COOKIE is set but deprecated
error: deprecated environment variables detected

Please use a configuration file instead; visit https://www.rabbitmq.com/configure.html to learn more
```

### PR DESCRIPTION
**Pull request by ajcasagrande**
I also updated the test code to check for this specific issue. If you run the updated tests against the current `master` it will fail due to the bug that this addresses.

Closes #338
Closes #341

### CODE REVIEW

File: wait/http.go [hunk 1]
```diff
@@ -184,12 +185,21 @@ func (ws *HTTPStrategy) WaitUntilReady(ctx context.Context, target StrategyTarge
 	address := net.JoinHostPort(ipAddress, strconv.Itoa(port.Int()))
 	endpoint := fmt.Sprintf("%s://%s%s", proto, address, ws.Path)
 
+	// cache the body into a byte-slice so that it can be iterated over multiple times
+	var body []byte
+	if ws.Body != nil {
+		body, err = io.ReadAll(ws.Body)
```
Comment (mdelapenya) [inline wait/http.go:191]: I think `io.ReadAll` was added in Go 1.16, right? https://pkg.go.dev/io/ioutil#ReadAll

**Review by mdelapenya — COMMENTED** [review]
@ajcasagrande thanks for taking your time to fix this, and the explanation on how to reproduce it.

I added a comment about the usage of a method that is failing in the test execution for Go 1.14.

Other than that, LGTM!

File: wait/http.go (hunk 1 above)
Comment (ajcasagrande) [reply wait/http.go]: Good catch. I have pushed the change.

**Comment by ajcasagrande** [comment]
Unit tests are failing due to #341. For now I have pinned the version of rabbitmq to `3.8.19` to band-aid the issue, since the latest `3.9.x` causes problems.

**Review by mdelapenya — APPROVED** [review]

**Pull request merged**